### PR TITLE
Remove default AppContext and cleanup globals

### PR DESCRIPTION
**Pull request by johnmccutchan**
This is an updated version of a closed pull request. This change keeps the context in the zone but removes the implicit global context. 

- [x] Remove the implicit global _defaultContext. This caught a bunch of places that we were assuming a context exists (and modifying it). We now require an AppContext to be constructed and the application code to call the runInZone method.

- [x] Move some global variables from globals.dart into their own files.

### CODE REVIEW

**Comment by johnmccutchan** [comment]
@devoncarew

File: packages/flutter_tools/lib/src/base/context.dart [hunk 1]
```diff
@@ -4,15 +4,13 @@
...
-
 typedef void ErrorHandler(dynamic error, StackTrace stackTrace);
 
 /// A singleton for application functionality. This singleton can be different
 /// on a per-Zone basis.
 AppContext get context {
-  AppContext currentContext = Zone.current['context'];
-  return currentContext == null ? _defaultContext : currentContext;
+  AppContext context = Zone.current['context'];
+  return context;
```
Comment (devoncarew) [inline packages/flutter_tools/lib/src/base/context.dart]: This function can use the shorthand syntax now (=>).

File: packages/flutter_tools/lib/src/build_info.dart [hunk 2]
```diff
@@ -118,6 +119,9 @@ HostPlatform getCurrentHostPlatform() {
 
 /// Returns the top-level build output directory.
 String getBuildDirectory() {
+  if (context == null) {
```
Comment (devoncarew) [inline packages/flutter_tools/lib/src/build_info.dart]: nit: no braces

File: packages/flutter_tools/test/analyze_duplicate_names_test.dart [hunk 3]
```diff
@@ -41,8 +39,6 @@ void main() {
         expect(testLogger.statusText, startsWith('Analyzing 2 files...\nNo analyzer warnings!'));
       });
 
-    }, overrides: <Type, dynamic>{
-      OperatingSystemUtils: os
```
Comment (devoncarew) [inline packages/flutter_tools/test/analyze_duplicate_names_test.dart]: Was this override not used?

Comment (johnmccutchan) [inline packages/flutter_tools/test/analyze_duplicate_names_test.dart]: The issue here is that the 'os' global reaches into the current context.

File: packages/flutter_tools/test/os_utils_test.dart [hunk 4]
```diff
@@ -21,16 +22,21 @@ void main() {
     });
 
     test('makeExecutable', () {
-      File file = new File(path.join(temp.path, 'foo.script'));
-      file.writeAsStringSync('hello world');
-      os.makeExecutable(file);
+      // Construct a context.
+      AppContext testContext = new AppContext();
```
Comment (johnmccutchan) [inline packages/flutter_tools/test/os_utils_test.dart]: The issue here is that the 'os' global reaches into the current context.

File: packages/flutter_tools/lib/src/build_info.dart (hunk 2 above)
Comment (johnmccutchan) [inline packages/flutter_tools/lib/src/build_info.dart]: Fixed.

File: packages/flutter_tools/test/analyze_duplicate_names_test.dart (hunk 3 above)
Comment (devoncarew) [inline packages/flutter_tools/test/analyze_duplicate_names_test.dart]: From looking into this, it looks like this override was not used.

**Comment by devoncarew** [comment]
lgtm

File: packages/flutter_tools/test/os_utils_test.dart (hunk 4 above)
Comment (johnmccutchan) [inline packages/flutter_tools/test/os_utils_test.dart]: Good point. Done,

**Pull request merged**